subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -1890,20 +1906,93 @@ static int do_write_op(struct lwm2m_engine_obj *obj,
 	}
 }
 
-static int get_observe_option(const struct zoap_packet *zpkt)
+static int get_option_int(const struct zoap_packet *zpkt, u8_t opt)
 {
 	struct zoap_option option = {};
 	u16_t count = 1;
 	int r;
 
-	r = zoap_find_options(zpkt, ZOAP_OPTION_OBSERVE, &option, count);
+	r = zoap_find_options(zpkt, opt, &option, count);
 	if (r <= 0) {
 		return -ENOENT;
 	}
 
 	return zoap_option_value_to_int(&option);
 }
 
+static int
+init_block_ctx(const u8_t *token, u8_t tkl, struct block_context **ctx)
+{
+	int i;
+	s64_t timestamp;
+
+	*ctx = NULL;
+	timestamp = k_uptime_get();
+	for (i = 0; i < NUM_BLOCK1_CONTEXT; i++) {
+		if (block1_contexts[i].tkl == 0) {
+			*ctx = &block1_contexts[i];
+			break;
+		}
+		if (timestamp-block1_contexts[i].timestamp >
+		    TIMEOUT_BLOCKWISE_TRANSFER) {
+			*ctx = &block1_contexts[i];
+			/* TODO: notify application for block
+			 * transfer timeout
+			 */
+			break;
+		}
+	}
+
+	if (*ctx == NULL) {
+		SYS_LOG_ERR("Cannot find free block context");
+		return -ENOMEM;
+	}
+
+	(*ctx)->tkl = tkl;
+	memcpy((*ctx)->token, token, tkl);
+#if defined(CONFIG_NET_L2_BLUETOOTH)
+	zoap_block_transfer_init(&(*ctx)->ctx, ZOAP_BLOCK_64, 0);
+#else
+	zoap_block_transfer_init(&(*ctx)->ctx, ZOAP_BLOCK_256, 0);
+#endif
```

Comment:
A JIRA ticket is created for this https://jira.zephyrproject.org/browse/ZEP-2416.
I suppose we will fix this later on.